src/main/java/me/hypherionmc/sdlinklib/discord/commands/OfflineWhitelist.java

```diff
@@ -169,16 +168,14 @@ protected void execute(CommandEvent event) {
                         if (!tables.isEmpty()) {
                             tables.forEach(SQLiteTable::delete);
 
-                            String nickName = (event.getMember().getNickname() == null || event.getMember().getNickname().isEmpty()) ? event.getAuthor().getName() : event.getMember().getNickname();
-                            if (pattern.matcher(nickName).matches()) {
-                                nickName = pattern.matcher(nickName).replaceAll("");
-                            }
-
-                            try {
-                                event.getMember().modifyNickname(nickName).queue();
-                            } catch (Exception e) {
-                                if (modConfig.generalConfig.debugging) {
-                                    e.printStackTrace();
+                            String nickName = event.getMember().getEffectiveName();
+                            if (pattern.matcher(nickName).find()) {
+                                try {
+                                    event.getMember().modifyNickname(null).queue();
```

Comment:
Please don't clear the nickname. The user could've changed their nickname themselves, after account linking and clearing it would result in some unhappy people

Reply:
The nickname is only cleared if it matches the Pattern, which would be only if they have the [MC: Username], before this change I made, the nickname was being changed always to the same nickname, which means: it wasn't doing anything to the nickname after the unlink (the method "matches" only work if it matches the whole text, which was never the case, cause even the RegEx was incorrect with the nickname pattern.